### PR TITLE
[New Rule] Azure External Guest User Invitation

### LINKED ISSUE #193: [New Rule] Azure External Guest User Invitation
## Description
Identifies an invitation to an external user in Azure Active Directory. Azure AD is extended to include Azure AD B2B collaboration, allowing you to invite people from outside your organization to be guest users in your cloud account. Unless there's a business need to provide guest access to any user, avoid creating guest users. Guest users could potentially be overlooked indefinitely leading to a potential vulnerability.

## Required Info

- **Eventing Sources:**
`filebeat-*`

- **Platforms**
`azure`

- **Target ECS Version:** `1.5.0`
- **New fields required in ECS for this?**
- **Related issues or PRs**

## Optional Info
- **References:**
https://docs.microsoft.com/en-us/azure/governance/policy/samples/cis-azure-1-1-0

## Syntax
```
event.module:azure and event.dataset:azure.auditlogs and event.category:AuditLogs and azure.auditlogs.operation_name:"Invite external user" and azure.auditlogs.properties.target_resources.0.display_name:guest and event.outcome:Success
```

## Example Data
image

### PR DESCRIPTION
**Pull request by bm11100**
## Issue(s)
resolves #193 

## Description

Identifies an invitation to an external user in Azure Active Directory (AD). Azure AD is extended to include collaboration, allowing you to invite people from outside your organization to be guest users in your cloud account. Unless there is a business need to provide guest access to any user, avoid creating guest users. Guest users could potentially be overlooked indefinitely leading to a potential vulnerability.

## Checklists
Use ~~strikethroughs~~ to remove items which are not applicable to this issue.

### For submitter
- [x] Verify all query fields co-exist in a single event source (*beats) or annotated otherwise
  -- Query field process.name.text doesn't currently exist in endgame-* eventing data, but does exist for winlogbeats, and is case insensitive.
- [x] Verify all query fields co-exist within a single ECS version and annotated the minimum version
  -- 1.4.0
- [ ] Verified the query detects the intended event(s)
  - detonate
  - search in _discover_
  - trigger as a custom signal
- [x] Create rule using `create-rule`
- [x] Verify and convert to standard linting (`toml-lint -f <rule-file>`)
- [x] Run tests using `pytest -x -v unit_tests` or `make test`
- [ ] Internal search to determine noise and FP

### For reviewers
- [ ] Verify existing rule for activity doesn't exist as a siem or endpoint rule (unless intentionally specified in issue)
- [ ] Verify all query fields co-exist in a single event source (*beats) or annotated otherwise
- [ ] Verify all query fields co-exist within a single ECS version and annotated the minimum version
- [ ] Internal search to determine noise and FP
- [ ] Verify metadata accuracy and spelling

### CODE REVIEW

File: rules/azure/initial_access_external_guest_user_invite.toml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+note = "The Azure Filebeat module must be enabled to use this rule."
+references = ["https://docs.microsoft.com/en-us/azure/governance/policy/samples/cis-azure-1-1-0"]
+risk_score = 21
+rule_id = "141e9b3a-ff37-4756-989d-05d7cbf35b0e"
+severity = "low"
+tags = ["Elastic", "Azure", "SecOps", "Continuous Monitoring", "Identity and Access"]
+type = "query"
+
+query = '''
+event.module:azure and event.dataset:azure.auditlogs and event.category:AuditLogs and azure.auditlogs.operation_name:"Invite external user" and azure.auditlogs.properties.target_resources.*.display_name:guest and event.outcome:Success
```
Comment (brokensound77) [inline rules/azure/initial_access_external_guest_user_invite.toml:36]: Is the `*` required in `azure.auditlogs.properties.target_resources.*.display_name:guest`?

Comment (bm11100) [reply rules/azure/initial_access_external_guest_user_invite.toml:36]: it's not required, but I wasn't sure if that would ever change. currently it has a 0 -  `azure.auditlogs.properties.target_resources.0.display_name`.  we discussed with another rule and it works with ES and passes tests, so we should be good, its not at the beginning or end. this is also going to crop up with GCP rules as many use versions, like v1 or v2, and we'll probably need to make the query `v*`

Comment (brokensound77) [reply rules/azure/initial_access_external_guest_user_invite.toml:36]: Yea I remember the convo but not the outcome. Sounds good to me - we will just need to keep an eye on the module too I suppose

File: rules/azure/initial_access_external_guest_user_invite.toml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ecs_version = ["1.6.0"]
+maturity = "production"
+updated_date = "2020/08/31"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies an invitation to an external user in Azure Active Directory (AD). Azure AD is extended to include
+collaboration, allowing you to invite people from outside your organization to be guest users in your cloud account.
+Unless there is a business need to provide guest access to any user, avoid creating guest users. Guest users could
```
Comment (threat-punter) [inline rules/azure/initial_access_external_guest_user_invite.toml:12]:
```suggestion
Unless there is a business need to provision guest access, it is best practice avoid creating guest users. Guest users could
```

File: rules/azure/initial_access_external_guest_user_invite.toml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+id = "T1078"
+name = "Valid Accounts"
+reference = "https://attack.mitre.org/techniques/T1078/"
+
+
+[rule.threat.tactic]
+id = "TA0001"
+name = "Initial Access"
+reference = "https://attack.mitre.org/tactics/TA0001/"
+
```
Comment (threat-punter) [inline rules/azure/initial_access_external_guest_user_invite.toml:52]:
```suggestion


[[rule.threat]]
framework = "MITRE ATT&CK"
[[rule.threat.technique]]
id = "T1078"
name = "Valid Accounts"
reference = "https://attack.mitre.org/techniques/T1078/"


[rule.threat.tactic]
id = "TA0003"
name = "Persistence"
reference = "https://attack.mitre.org/tactics/TA0003/"
```

What do you think about adding the same technique under the Persistence tactic too?

**Review by threat-punter — APPROVED** [review]

File: rules/azure/initial_access_external_guest_user_invite.toml (hunk 3 above)
Comment (bm11100) [reply rules/azure/initial_access_external_guest_user_invite.toml]: sounds good to me

**Pull request merged**